cmd/nsp/main.go

```diff
@@ -141,3 +131,34 @@ func main() {
 
 	<-ctx.Done()
 }
+
+func CreateTargetRegistryServer(ctx context.Context, config *Config) (nspAPI.TargetRegistryServer, error) {
+	sqlr, err := sqliteRegistry.New(config.Datasource)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable create sqlite registry")
+	}
+	keepAliveRegistry, err := keepAliveRegistry.New(
+		keepAliveRegistry.WithRegistry(sqlr),
+		keepAliveRegistry.WithTimeout(config.EntryTimeout),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable create keepalive registry")
+	}
+	return nsp.NewServer(
+		nspRegistry.NewServer(keepAliveRegistry),
```

Comment:
Separation of nspRegistry and watchhandler seems forced to me. They both rely on the same keepAliveRegistry.